docs/src/main/asciidoc/config-extending-support.adoc

```diff
@@ -0,0 +1,338 @@
+import io.smallrye.config.PropertiesConfigSource;
+
+public class URLConfigSourceFactory implements ConfigSourceFactory {
+    @Override
+    public Iterable<ConfigSource> getConfigSources(final ConfigSourceContext context) {
+        final ConfigValue value = context.getValue("config.url");
+        if (value == null || value.getValue() == null) {
+            return Collections.emptyList();
+        }
+
+        try {
+            return Collections.singletonList(new PropertiesConfigSource(new URL(value.getValue())));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public OptionalInt getPriority() {
+        return OptionalInt.of(290);
+    }
+}
+----
+
+And registration in:
+
+.META-INF/services/io.smallrye.config.ConfigSourceFactory
+[source,properties]
+----
+org.acme.config.URLConfigSourceFactory
+----
+
+By implementing `io.smallrye.config.ConfigSourceFactory`, a list of `ConfigSource` may be provided via the
+`Iterable<ConfigSource> getConfigSources(ConfigSourceContext context)` method. The `ConfigSourceFactory` may also
+assign a priority by overriding the default method `OptionalInt getPriority()`. This is only used to sort multiple
+factories during initialization. Once all the factories are initialized, the provided list of `ConfigSource` will use
+their own ordinal and sorted with all the other sources available in the `Config` instance.
+
+When the Factory is initializing, the provided `ConfigSourceContext` may call the method
+`ConfigValue getValue(String name)`. This method lookups configuration names in all `ConfigSource` that were already
+initialized by the `Config` instance, including sources with lower ordinals than the ones defined in the
+`ConfigSourceFactory`. The `ConfigSource` list provided by a `ConfigSourceFactory` is not taken into considerations to
+configure other sources produced by a lower priority `ConfigSourceFactory`.
+
+[[custom-converters]]
+== Custom Converters
+
+It is possible to create a Custom Converter type as specified by
+link:https://github.com/eclipse/microprofile-config/blob/master/spec/src/main/asciidoc/converters.asciidoc#adding-custom-converters[MicroProfile Config Custom Converters].
+
+A custom Converter requires an implementatin of `org.eclipse.microprofile.config.spi.Converter<T>`. Each implementation
+requires registration via the https://docs.oracle.com/javase/8/docs/api/java/util/ServiceLoader.html[ServiceLoader]
+mechanism, either in `META-INF/services/org.eclipse.microprofile.config.spi.Converter`. Consider:
+
+[source,java]
+----
+package org.acme.config;
+
+public class MicroProfileCustomValue {
+
+    private final int number;
+
+    public MicroProfileCustomValue(int number) {
+        this.number = number;
+    }
+
+    public int getNumber() {
+        return number;
+    }
+}
+----
+
+The corresponding converter will look like the one below. Please note that the custom converter class must be `public`
```

Comment:
"will look similar to the one below", or "will look something like the one below".